lib/bolt/util.rb

```diff
@@ -344,8 +344,8 @@ def validate_file(type, path, allow_dir = false)
 
         if !stat.readable?
           raise Bolt::FileError.new("The #{type} '#{path}' is unreadable", path)
-        elsif !stat.file? && (!allow_dir || !stat.directory?)
-          expected = allow_dir ? 'file or directory' : 'file'
+        elsif !(stat.file? || stat.pipe?) && (!allow_dir || !stat.directory?)
```

Comment:
I think in the spirit of "everything is a file", we can change this check to simply `elsif !allow_dir && stat.directory?`. What we really care about is just that we can read from it like a file, and the closest thing to that check we have is just that it's not a directory.

Similarly, it's probably simplest to still just report we expect it to be a "file or directory".

Reply:
makes sense to me, I updated the pull request

Reply:
@nicklewis can you take another look when you have a moment?

Reply:
Nick will be out for a bit, I think this is a change that we can get in. I just allowed some tests to run on this and some other checks. We will figure out a path forward for getting this in and released.

Reply:
@donoghuc thanks!

Reply:
@donoghuc some of the checks flagged errors, but some of them seem unrelated to this commit. Also, happy to add a release note, if desired?

Reply:
@donoghuc any chance you could take a look at the CI failures?